lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -1168,6 +1168,55 @@ getSharedEncIfAllUsersAreDotEnc(Value val, bool &incompatible) {
   return attr;
 }
 
+static Type getNewType(Type type, Attribute encoding) {
+  RankedTensorType tensorType = cast<RankedTensorType>(type);
+  return RankedTensorType::get(tensorType.getShape(),
+                               tensorType.getElementType(), encoding);
+}
+
+Operation *convertOpEncoding(Attribute encoding, Operation *op) {
```

Comment:
This function does not touch operands if they are of shared encoding but the name of this function is fairly general and we cannot tell that. Maybe find a better name for this function, or at least update the documentation to clarify?